### PR TITLE
remove dependency on boost in chrono.hpp

### PR DESCRIPTION
**Pull request by uyha**

### CODE REVIEW

**Comment by uyha** [comment]
@redboltz I have no idea why this just fails randomly for the chrono test cases, could you have the look?

**Comment by redboltz** [comment]
> @redboltz I have no idea why this just fails randomly for the chrono test cases, could you have the look?

Hmm, it this also happend the HEAD ac062e28cbf80db345800b6e113dea8ab0055499 ?
Or after your modification is applied?

**Comment by uyha** [comment]
fixed after i pushed 😅 , i forgot to remove the preprocess guard in type.hpp

File: include/msgpack/v1/adaptor/cpp11/chrono.hpp [hunk 1]
```diff
@@ -30,6 +27,97 @@ MSGPACK_API_VERSION_NAMESPACE(v1) {
...
 
+namespace detail {
+template<typename Target,
+         typename Source,
+         bool target_is_signed = std::is_signed<Target>::value,
+         bool source_is_signed = std::is_signed<Source>::value,
+         typename = typename std::enable_if<
+                      std::is_integral<Target>::value &&
+                      std::is_integral<Source>::value
+                    >::type>
```
Comment (redboltz) [inline include/msgpack/v1/adaptor/cpp11/chrono.hpp]:
```suggestion
template <
    typename Target,
    typename Source,
    bool target_is_signed = std::is_signed<Target>::value,
    bool source_is_signed = std::is_signed<Source>::value,
    typename = typename std::enable_if<
        std::is_integral<Target>::value &&
        std::is_integral<Source>::value
    >::type
>
```

Just coding style request:

1. Please add one whitespace between `template` and `<`.
2. Please use the same rule as (non template parameters) for template parameters.
3. Please set the same level `xxxx <` and `>`.

In addition, please use the same rule that I requested at the #1075.

**Review by redboltz — CHANGES_REQUESTED** [review]

File: include/msgpack/v1/adaptor/cpp11/chrono.hpp (hunk 1 above)
Comment (uyha) [reply include/msgpack/v1/adaptor/cpp11/chrono.hpp]: I don't understand the 2nd point, could you give an example?

Comment (redboltz) [reply include/msgpack/v1/adaptor/cpp11/chrono.hpp]: Ok, 2nd point.

# For non template function parameters:

## OK

```cpp
ret_type func_name(
    int param1,
    char* param2,
    some_type const& param3
) {
    // ...
}
```

```cpp
ret_type func_name(
    int param1,
    char* param2,
    some_type const& param3) {
    // ...
}
```

## NG

```cpp
ret_type func_name(int param1,
                   char* param2,
                   some_type const& param3
) {
    // ...
}
```

Ok, 2nd point.

# Fortemplate function parameters:


## OK

```cpp
template <typename T, typename U>
ret_type func_name(...) {
}
```

```cpp
template <
    typename T,
    typename U
>
ret_type func_name(...) {
}
```
## NG

```cpp
template <typename T,
          typename U
>
ret_type func_name(...) {
}
```

```cpp
template <typename T,
          typename U
         >
ret_type func_name(...) {
}
```

Comment (uyha) [reply include/msgpack/v1/adaptor/cpp11/chrono.hpp]: got it, thank you

File: include/msgpack/v1/adaptor/cpp11/chrono.hpp [hunk 2]
```diff
@@ -30,6 +27,113 @@ MSGPACK_API_VERSION_NAMESPACE(v1) {
...
 
+namespace detail {
+template <
+    typename Target,
+    typename Source,
+    bool target_is_signed = std::is_signed<Target>::value,
+    bool source_is_signed = std::is_signed<Source>::value,
+    typename = typename std::enable_if<
+                  std::is_integral<Target>::value &&
+                  std::is_integral<Source>::value
```
Comment (redboltz) [inline include/msgpack/v1/adaptor/cpp11/chrono.hpp]:
```suggestion
                   std::is_integral<Target>::value &&
                   std::is_integral<Source>::value
```

File: include/msgpack/v1/adaptor/cpp11/chrono.hpp [hunk 3]
```diff
@@ -30,6 +27,113 @@ MSGPACK_API_VERSION_NAMESPACE(v1) {
...
+          } {}
+    bool value;
+};
+
+template <
+    typename Target,
+    typename Source,
+    typename = typename std::enable_if<
+                  std::is_integral<Target>::value &&
+                  std::is_integral<Source>::value
```
Comment (redboltz) [inline include/msgpack/v1/adaptor/cpp11/chrono.hpp]:
```suggestion
                   std::is_integral<Target>::value &&
                   std::is_integral<Source>::value
```

**Review by redboltz — CHANGES_REQUESTED** [review]
I noticed that some of indents are 3.

File: include/msgpack/v1/adaptor/cpp11/chrono.hpp (hunk 2 above)
Comment (uyha) [reply include/msgpack/v1/adaptor/cpp11/chrono.hpp]: yeah, mistakes from doing it manually

**Review by redboltz — APPROVED** [review]

**Comment by redboltz** [comment]
Merged. Thank you!

**Pull request merged**